generator/restriction_collector.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "indexer/routing.hpp"
+
+#include "std/functional.hpp"
+#include "std/limits.hpp"
+#include "std/string.hpp"
+#include "std/unordered_map.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+/// This class collects all relations with type restriction and save feature ids of
+/// their road feature in text file for using later.
+class RestrictionCollector
+{
+  friend void UnitTest_RestrictionTest_ValidCase();
+  friend void UnitTest_RestrictionTest_InvalidCase();
+  friend void UnitTest_RestrictionTest_ParseRestrictions();
+  friend void UnitTest_RestrictionTest_ParseFeatureId2OsmIdsMapping();
+
+public:
+  /// \brief Addresses a link in vector<Restriction>.
+  struct LinkIndex
+  {
+    LinkIndex(size_t restrictionNumber, size_t linkNumber)
+      : m_restrictionNumber(restrictionNumber), m_linkNumber(linkNumber)
```

Comment:
It means link number. For the time being it could be 0 or 1. With the help of LinkIndex we can point at outgoing (linkNumber == 0) and ingoing (linkNumber == 1) restriction links for every restriction (restrictionNumber).